### ISSUE TITLE
TypeScript complains about .use(AxiosLogger.requestLogger)

### ISSUE THREAD

**Issue by ivanakcheurov** [hg-pyun/axios-logger#131]
Typescript shows squiggly lines for this code:
```ts
instance.interceptors.request.use(AxiosLogger.requestLogger, AxiosLogger.errorLogger);
```
![image](https://github.com/hg-pyun/axios-logger/assets/5347026/189b1fa4-0ca5-497e-9ef0-36fbe53bd0ae)
Error text:
```
Argument of type '(request: AxiosRequestConfig<any>, config?: RequestLogConfig | undefined) => AxiosRequestConfig<any>' is not assignable to parameter of type '(value: InternalAxiosRequestConfig<any>) => InternalAxiosRequestConfig<any> | Promise<InternalAxiosRequestConfig<any>>'.
  Type 'AxiosRequestConfig<any>' is not assignable to type 'InternalAxiosRequestConfig<any> | Promise<InternalAxiosRequestConfig<any>>'.
    Type 'AxiosRequestConfig<any>' is not assignable to type 'InternalAxiosRequestConfig<any>'.
      Types of property 'headers' are incompatible.
        Type 'AxiosHeaders | (Partial<RawAxiosHeaders & { Accept: AxiosHeaderValue; "Content-Length": AxiosHeaderValue; "User-Agent": AxiosHeaderValue; "Content-Encoding": AxiosHeaderValue; Authorization: AxiosHeaderValue; } & { ...; }> & Partial<...>) | undefined' is not assignable to type 'AxiosRequestHeaders'.
          Type 'undefined' is not assignable to type 'AxiosRequestHeaders'.
            Type 'undefined' is not assignable to type 'Partial<RawAxiosHeaders & { Accept: AxiosHeaderValue; "Content-Length": AxiosHeaderValue; "User-Agent": AxiosHeaderValue; "Content-Encoding": AxiosHeaderValue; Authorization: AxiosHeaderValue; } & { ...; }>'.ts(2345)
```
**Desktop (please complete the following information):**
Ubuntu based.
```bash
$ lsb_release -a
No LSB modules are available.
Distributor ID: Tuxedo
Description:    TUXEDO OS 2
Release:        22.04
Codename:       jammy

$ uname -r
6.2.0-10007-tuxedo

$ node -v
v16.20.1

$ npx tsc -v
Version 5.0.4
```

**Comment by hg-pyun** [comment]
I will fix this PR -> #135

**Comment by hg-pyun** [comment]
https://github.com/hg-pyun/axios-logger/pull/136

**Comment by hg-pyun** [comment]
Release 2.7.0 version. Please Check it. 🙇‍♂️

**Comment by alainfonhof** [comment]
I'm still getting a TypeError on version 2.7.0:
```
Type 'undefined' is not assignable to type 'AxiosRequestHeaders'.
            Type 'undefined' is not assignable to type 'Partial<RawAxiosHeaders & { Accept: AxiosHeaderValue; "Content-Length": AxiosHeaderValue; "User-Agent": AxiosHeaderValue; "Content-Encoding": AxiosHeaderValue; Authorization: AxiosHeaderValue; } & { 'Content-Type': ContentType; }>'.ts(2345)
```

**Comment by cmaster11** [comment]
Same issue here with 2.7.0

```
TS2345: Argument of type (req: InternalAxiosRequestConfig<any>) => AxiosRequestConfig<any> is not assignable to parameter of type (value: InternalAxiosRequestConfig<any>) => InternalAxiosRequestConfig<any> | Promise<InternalAxiosRequestConfig<any>>
	Type  AxiosRequestConfig<any>  is not assignable to type nternalAxiosRequestConfig<any> | Promise<InternalAxiosRequestConfig<any>>
		Type  AxiosRequestConfig<any>  is not assignable to type InternalAxiosRequestConfig<any> 
		Types of property  headers  are incompatible.
			Type AxiosHeaders | (Partial<RawAxiosHeaders & {   Accept: AxiosHeaderValue;   'Content-Length': AxiosHeaderValue;   'User-Agent': AxiosHeaderValue;   'Content-Encoding': AxiosHeaderValue;   Authorization: AxiosHeaderValue; } & { ...; }> & Partial<...>) | undefined is not assignable to type  AxiosRequestHeaders 
				Type  undefined  is not assignable to type  AxiosRequestHeaders 
					Type  undefined  is not assignable to type Partial<RawAxiosHeaders & {   Accept: AxiosHeaderValue;   'Content-Length': AxiosHeaderValue;   'User-Agent': AxiosHeaderValue;   'Content-Encoding': AxiosHeaderValue;   Authorization: AxiosHeaderValue; } & { ...; }>
```

**Comment by hg-pyun** [comment]
@alainfonhof @cmaster11 I will check it. Thank you report.

**Comment by hg-pyun** [comment]
@alainfonhof @cmaster11 Can you find out which version of axios and typescript you used?

**Comment by cmaster11** [comment]
@hg-pyun from lock file

```
axios-logger@^2.7.0:
  version "2.7.0"
axios@^1.4.0:
  version "1.4.0"
typescript@^5.1.6:
  version "5.1.6"
```

**Comment by leechoohyoung** [comment]
same here. 

  axios:
    specifier: ^1.4.0
    version: 1.4.0
  axios-logger:
    specifier: ^2.7.0
    version: 2.7.0
  typescript:
    specifier: ^4.9.5
    version: 4.9.5

**Comment by imincode** [comment]
Just trying to start using this library and got the same error. Any update?

**Comment by marco-merola-ef** [comment]
This should work with axios-logger 2.7.0 and axios 1.5.0
```
import axios, { Axios, InternalAxiosRequestConfig } from 'axios';
import * as AxiosLogger from 'axios-logger';

. . .

        this.client = axios.create({
            . . .
        });

        this.client.interceptors.request.use(this.onFulfilledInterceptor, AxiosLogger.errorLogger);

        this.client.interceptors.response.use(AxiosLogger.responseLogger, AxiosLogger.errorLogger);

. . .

          private onFulfilledInterceptor = (config: InternalAxiosRequestConfig) => {
             AxiosLogger.requestLogger(config);
             return config;
          };
```

**Comment by bforbis** [comment]
Just cast to `any` to fix the error for now. If your eslint complains, also tell it to go away

```ts
// eslint-disable-next-line @typescript-eslint/no-explicit-any
client.interceptors.request.use( AxiosLogger.requestLogger as any, AxiosLogger.errorLogger );
```

**Comment by hg-pyun** [comment]
I merged #147 and pulished version 2.7.1
